### PR TITLE
diff: document --hide-missing option

### PR DESCRIPTION
**Pull request by pmrowla**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Documentation changes for https://github.com/iterative/dvc/pull/4530

### CODE REVIEW

File: content/docs/command-reference/diff.md [hunk 1]
```diff
@@ -33,6 +33,14 @@ sections below for more details.
 `dvc diff` does not have an effect when the repository is not tracked by Git,
 for example when `dvc init` was used with the `--no-scm` option.
 
+By default, when comparing the current workspace to a Git commit, `dvc diff`
+includes a `not in cache` status for DVC-tracked files and directories which are
```
Comment (jorgeorpinel) [inline content/docs/command-reference/diff.md:37]: I cannot reproduce this.

```console
λ echo data > data
λ dvc add data
...
λ dvc diff
Added:
    data

files summary: 1 added, 0 deleted, 0 modified

λ rm data
λ dvc diff  # No output
```

Comment (jorgeorpinel) [reply content/docs/command-reference/diff.md:37]: (DVC 1.6.1 on Windows)

File: content/docs/command-reference/diff.md [hunk 2]
```diff
@@ -33,6 +33,14 @@ sections below for more details.
...
+neither in the workspace nor the local cache (indicating that they most likely
+need to be `dvc pull`ed into the workspace). The `--hide-missing` option can be
+used to suppress this status and only print files and directories which have
+been expliclity added, modified or deleted in the workspace. This status will be
+automatically suppressed when comparing two Git commits.
```
Comment (shcheklein) [inline content/docs/command-reference/diff.md:42]: My 2cs on this - unless it's some critical option, I would try to make majority of this description into the Options section below (it's not scalable to describe every option in the main section, also the purpose of it is to describe general behavior, edge cases, etc. So, it's fine to reference an option, but probably not provide full description)

File: content/docs/command-reference/diff.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/diff.md:37]: > most likely need to be `dvc pull`ed

Prob no need for this info in the `diff` ref.

File: content/docs/command-reference/diff.md (hunk 2 above)
Comment (jorgeorpinel) [inline content/docs/command-reference/diff.md:42]: We can just explain the option in the Options section I think. Like `--show-md` which isn't mentioned in the Description. These are uncommon options, I think.

**Review by jorgeorpinel — CHANGES_REQUESTED** [review]
Thanks for this @pmrowla 

Other than the comments above, it's kind of confusing to mix the concept of _status_ with diff. Are these the same states that `dvc status` handles for example? The explanation can get complicated quickly. It would be best to keep that term out of the text in this ref. if possible

> - [x] BTW we should remove it from the `--show-md` option desc. probably. I can do that later...

File: content/docs/command-reference/diff.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/diff.md:42]: I see Ivan already mentioned this in https://github.com/iterative/dvc.org/pull/1758#pullrequestreview-484504526 above. Resolving here.

File: content/docs/command-reference/diff.md (hunk 1 above)
Comment (pmrowla) [reply content/docs/command-reference/diff.md:37]: PR was just merged recently, and won't show up until https://github.com/iterative/dvc/releases/tag/untagged-82c123df4a6da19799cf is released

Comment (jorgeorpinel) [reply content/docs/command-reference/diff.md:37]: Yeah I saw that PR after commenting here. Unfortunately there's some problems with the development version on Windows though so I can't check this :/ could you post a brief console example?

Comment (pmrowla) [reply content/docs/command-reference/diff.md:37]: https://asciinema.org/a/thHUinCl2BD3rAMcU5cEYZufO

**Comment by pmrowla** [comment]
Still thinking about how to best re-word the docs for this without mixing/confusing `dvc status` terminology, I'll push an update later today or early tomorrow

File: content/docs/command-reference/diff.md [hunk 3]
```diff
@@ -26,9 +26,9 @@ name, Git commit hash, etc.
 It defaults to comparing the current workspace and the last commit (`HEAD`), if
 arguments `a_rev` and `b_rev` are not specified.
 
-Options `--show-json` and `--show-hash` can be used to modify format and details
-of the output produced. See the [Options](#options) and [Examples](#examples)
-sections below for more details.
+Options `--show-json`, `--show-hash` and `--hide-missing` can be used to modify
+format and details of the output produced. See the [Options](#options) and
```
Comment (jorgeorpinel) [inline content/docs/command-reference/diff.md:30]:
```suggestion
the format and details of the output produced. See the [Options](#options) and
```

File: content/docs/command-reference/diff.md [hunk 4]
```diff
@@ -26,9 +26,9 @@ name, Git commit hash, etc.
 It defaults to comparing the current workspace and the last commit (`HEAD`), if
 arguments `a_rev` and `b_rev` are not specified.
 
-Options `--show-json` and `--show-hash` can be used to modify format and details
-of the output produced. See the [Options](#options) and [Examples](#examples)
-sections below for more details.
+Options `--show-json`, `--show-hash` and `--hide-missing` can be used to modify
```
Comment (jorgeorpinel) [inline content/docs/command-reference/diff.md:29]:
```suggestion
Options `--show-json`, `--show-hash`, and `--hide-missing` can be used to modify
```

Comment (jorgeorpinel) [reply content/docs/command-reference/diff.md:29]: Not sure we want to mention hide-missing here though. Maybe just

> `--show-json` and `--show-hash` can be used to modify the format of the output

File: content/docs/command-reference/diff.md [hunk 5]
```diff
@@ -52,6 +52,13 @@ for example when `dvc init` was used with the `--no-scm` option.
 - `--show-md` - print the list of files and directories with their status in the
   Markdown table format.
 
+- `--hide-missing` - do not print information about missing files. By default,
+  when comparing the current workspace to a Git commit, `dvc diff` includes a
+  list of dvc-tracked files which are missing in both the workspace and local
+  cache. This option can be used to suppress this information and only print
+  files and directories which have been expliclity added, modified or deleted in
+  the workspace. This option does nothing when comparing two Git commits.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/diff.md:60]: Thanks! A bit repetitive. How about

> - do not list files missing both from workspace and cache (`not in cache`). With this options, only files and directories which have been explicitly added, modified, or deleted are listed. This option has no effect when comparing two Git commits.

File: content/docs/command-reference/diff.md (hunk 4 above)
Comment (pmrowla) [reply content/docs/command-reference/diff.md]: reverted these changes to just omit `--hide-missing` from this part of the `dvc diff` docs

File: content/docs/command-reference/diff.md (hunk 5 above)
Comment (pmrowla) [reply content/docs/command-reference/diff.md]: Latest revision uses your suggestion here

File: content/docs/command-reference/diff.md (hunk 4 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/diff.md]: Thanks. We probably should say "and details" now though... I'll update it separately.

**Review by jorgeorpinel — APPROVED** [review]
I think we've got it now. Thanks for bearing with me @pmrowla

**Pull request merged**